### ISSUE TITLE
CSS Parser does not preserve styles used inside templates script[type=text/plain]

### ISSUE THREAD

**Issue by spasibych** [ampproject/amp-wp#4276]
## Bug Description

The plugin implements the removal of unused CSS rules on the page. This works well for all page content, including `<template type="amp-mustache">...</template>`, but does not work with `<script type="text/plain" template="amp-mustache">...</script>`.

## Expected Behaviour

Preserve CSS rules for elements inside `<script type="text/plain" template="amp-mustache">...</script>`.

## Steps to reproduce

1. Add code to AMP page:
```html
<style>
    h2.one { color: green }
    h2.two { color: red }
</style>
<template type="amp-mustache">
    <h2 class="one">One</h2>
</template>
<script type="text/plain" template="amp-mustache">
    <h2 class="two">Two</h2>
</script>
```
2. View the resulting source code for an AMP page.
3. See content `<style amp-custom>...</style>` in page head where there is a rule ` h2.one{color:green}` but there is no rule `h2.two{color:red}`.

## Additional context

 - WordPress version: 5.3.2
 - Plugin version: 1.4.2
 - AMP plugin template mode: Website; Standard; Serve all templates as AMP regardless of what is being queried.
 - PHP version: 7.3.11

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
I believe this may be resolved by fixing #4254.

**Comment by spasibych** [comment]
@westonruter thanks!

This is especially important if full support of `<!--{{#foo}}-->` is not implemented (issue [#26656](https://github.com/ampproject/amphtml/issues/26656)).

**Comment by schlessera** [comment]
> I believe this may be resolved by fixing #4254.

Just verified that this is indeed fixed via the PR that @kienstra did for #4254 :

<img width="370" alt="Image 2020-03-02 at 6 42 36 AM" src="https://user-images.githubusercontent.com/83631/75648694-1f922800-5c51-11ea-841f-a855f089184b.png">

I'll add a "Fixes #4276" to that PR.

**Comment by csossi** [comment]
Verified in QA:

</header>
--
  |  
  | <div class="post-inner thin ">
  |  
  | <div class="entry-content">
  |  
  |  
  | <style>
  | h2.one { color: green }
  | h2.two { color: red }
  | </style>
  | <template type="amp-mustache">
  | <h2 class="one">One</h2>
  | </template>
  | <script type="text/plain" template="amp-mustache">
  | <h2 class="two">Two</h2>
  | </script>

**Comment by spasibych** [comment]
Bug is present with plugin version 1.5 (1.5.0 and 1.5.1).

**Comment by westonruter** [comment]
Bizzare. We did fix this. But I'm not seeing it now.

**Comment by westonruter** [comment]
@spasibych Here’s a 1.5.2-RC1 pre-release build with the fix for you to test: [amp.zip](https://github.com/ampproject/amp-wp/files/4429512/amp.zip) (v1.5.2-RC1-20200403T193031Z-3102c5dea)

**Comment by westonruter** [comment]
We've just published the 1.5.2 release to WordPress.org: https://wordpress.org/plugins/amp/

You can update now via the WordPress admin.

Release notes: https://github.com/ampproject/amp-wp/releases/tag/1.5.2

**Comment by grapheon** [comment]
@westonruter now everything is fine, thank you very much!
